### PR TITLE
SCP-3132 Get total funds through WBE (part 1 - backend API)

### PR DESCRIPTION
**Pull request by hrajchert**
This PR refactors the current backend to modularize endpoints.
It also adds a new endpoint for get the total funds from the WBE.

Note that the `restoreWallet` is placed under a `CentralizedTestnet` module, indicating that this endpoint is only useful in that architecture, but the getTotalFunds is placed in plain Wallet, as its going to be useful for this architecture and also the full node one.

The previous `restoreWallet` endpoint is now under
```
POST /api/wallet/centralized-testnet/restore
  {  "getMnemonicPhrase" : ["word1", "word2", "...", "word24"]
  ,  "getPassphrase": "something"
  , "getWalletName": "my-wallet"
  }
```
The new `getTotalFunds is under

```
GET /api/wallet/<wallet-id>/get-total-funds
```

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [X] Commit sequence broadly makes sense
    - [X] Key commits have useful messages
    - [X] Relevant tickets are mentioned in commit messages
    - [X] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [X] Self-reviewed the diff
    - [X] Useful pull request description
    - [X] Reviewer requested

### CODE REVIEW

File: marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE DataKinds             #-}
```
Comment (hrajchert) [inline marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs:1]: The logic on this file hasn't change basically... only two minor changes commented below

File: marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                                                                        ResponseF (responseBody), client)
+import           Text.Regex                                            (Regex)
+import qualified Text.Regex                                            as Regex
+import qualified Wallet.Emulator.Wallet                                as Pab.Wallet
+
+handlers ::
+    MonadIO m =>
+    MonadReader Env m =>
+    ServerT API m
+handlers = restoreWallet
```
Comment (hrajchert) [inline marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs:49]: Added this as a placeholder for all CentralizedTestnet endpoints

File: marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                Nothing       ->  do
+                    putStrLn ("restoreWallet failed" :: Text)
+                    putStrLn $ "Error: " <> (show err :: Text)
+                    pure $ Left err
+        Left err -> do
+            -- FIXME: Define a better logging mechanism
+            putStrLn ("restoreWallet failed" :: Text)
+            putStrLn $ "Error: " <> (show err :: Text)
+            pure $ Left err
+        Right WBE.ApiWallet{WBE.id = WBE.ApiT walletId} -> do
```
Comment (hrajchert) [inline marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs:143]: I modified the pattern match expression from this

```haskell
Right (WBE.ApiWallet (WBE.ApiT walletId) _ _ _ _ _ _ _ _) -> do
```
As I think the new one is clearer.

File: marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Types.hs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE DeriveAnyClass     #-}
```
Comment (hrajchert) [inline marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Types.hs:1]: I got these types from `marlowe-dashboard-server/src/Marlowe/Run/Webserver/Types.hs`... nothing new here

File: marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs [hunk 5]
```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE OverloadedStrings     #-}
```
Comment (bwbush) [inline marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs:4]: If you remove `{-# LANGUAGE OverloadedStrings #-}`, then you can also remove the several `:: Text` annotations. The protolude's `show` and `putStrLn` couldn't handle the ambiguity in string overloading, so if you make the strings into actual `String` types, then the ambiguity is removed.

**Review by bwbush — APPROVED** [review]
- [x] I reviewed all code changes.
- [x] I manually tested the `api/wallet/.../get-total-funds` endpoint using `curl`. It reports correct results for (1) empty wallet, (2) ada-only wallet, (3) wallet with ada and tokens.
- [x] I manually tested the `api/wallet/centralized-testnet/restore` endpoint using `curl`.

Even at this stage of development, it would be useful if the error messages distinguished between bad post content (e.g., serialization) versus invalid endpoints.

File: marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs (hunk 5 above)
Comment (hrajchert) [reply marlowe-dashboard-server/src/Marlowe/Run/Webserver/Wallet/CentralizedTestnet/Server.hs:4]: Nice catch! it worked fine :). I'll add it as part of the next PR so we don't have to re-approve this one

**Comment by hrajchert** [comment]
@bwbush I created task SCP-3207 to solve the error reporting

**Pull request merged**